okhttp/src/main/java/com/squareup/okhttp/CacheControl.java

```diff
@@ -41,9 +41,11 @@
   private final boolean onlyIfCached;
   private final boolean noTransform;
 
+  volatile String headerValue; // Lazily computed, if absent.
```

Comment:
Is volatile necessary here? I fear this might be a pessimization with `volatile`, since we're invalidating CPU caches. (And we can implement this correctly without it.)